coriolis_openstack_utils/resource_utils/routers.py

```diff
@@ -0,0 +1,143 @@
+                     'availability_zone_hints'}
+
+    body = {k: v for k, v in router.items() if k in relevant_keys}
+    body['availability_zone_hints'] += router['availability_zones']
+    # eliminate any possible duplicates
+    body['availability_zone_hints'] = list(
+        set(body['availability_zone_hints']))
+    ext_gateway_info = router['external_gateway_info']
+
+    src_ext_subnet_ids = set()
+    for fixed_ip in ext_gateway_info['external_fixed_ips']:
+        src_ext_subnet_ids.add(fixed_ip['subnet_id'])
+    # mapped with external_network_name_map
+    ext_networks_ids = [subnets.get_subnet(source_client, subnet_id)[
+        'network_id'] for subnet_id in src_ext_subnet_ids]
+    ext_network_names = [networks.get_network(source_client, network_id)[
+        'name'] for network_id in ext_networks_ids]
+
+    # mapped with new_subnet_name and new_network_name
+    src_subnet_ids = set()
+    for port in source_client.neutron.list_ports(
+            device_id=router['id'])['ports']:
+        for fixed_ip in port['fixed_ips']:
+            src_subnet_ids.add(fixed_ip['subnet_id'])
+
+    src_subnet_ids = src_subnet_ids - src_ext_subnet_ids
+
+    src_subnets = [subnets.get_subnet(source_client, subnet_id)
+                   for subnet_id in src_subnet_ids]
+
+    # adding both network name and subnet name the chance of a collision on
+    # destination is greatly reduced
+    src_subnets = [{'subnet_name': subnet['name'],
+                    'network_name': networks.get_network(
+                        source_client, subnet['network_id'])['name']}
+                   for subnet in src_subnets]
+
+    return {'source_name': router['name'],
+            'migration_body': body,
+            'src_ext_net_names': ext_network_names,
+            'src_subnets': src_subnets}
+
+
+def create_router(destination_client, migr_info):
+    body = migr_info['migration_body']
+    src_router_name = migr_info['source_name']
+    new_format = CONF.destination.new_router_name_format
+    body['name'] = new_format % {
+        "original": src_router_name}
+    router_id = destination_client.neutron.create_router(
+        {'router': body})['router']['id']
+
+    src_ext_net_names = migr_info['src_ext_net_names']
+    src_subnets = migr_info['src_subnets']
+
+    external_network_map = CONF.destination.external_network_name_map
+    dest_ext_net_names = [external_network_map.get(net_name, net_name)
+                          for net_name in src_ext_net_names]
+
+    dest_ext_net_ids = [networks.get_network(
+        destination_client, net_name)['id'] for net_name in dest_ext_net_names]
+    for net_id in dest_ext_net_ids:
+        LOG.info(
+            "Adding external network %s gateway to router %s"
+            % (net_id, router_id))
+        destination_client.neutron.add_gateway_router(
+            router_id, {'network_id': net_id})
+
+    new_net_name = CONF.destination.new_network_name_format
```

Comment:
Please clearly define these as `new_net_name_format` and `new_subnet_name_format` (I initially thought you were forgetting to format in the names.)